commands/init.ts

```diff
@@ -73,95 +76,125 @@ const personalAccessKeyConfigCreationFlow = async (env, account) => {
     logError(e);
   }
   return updatedConfig;
-};
+}
 
-const oauthConfigCreationFlow = async env => {
-  const configData = await promptUser(OAUTH_FLOW);
-  const accountConfig = {
+async function oauthConfigCreationFlow(env: Environment) {
+  const configData = await promptUser<OauthPromptResponse>(OAUTH_FLOW);
+  const accountConfig: OAuth2ManagerAccountConfig = {
     ...configData,
     env,
   };
   await authenticateWithOauth(accountConfig);
-  updateDefaultAccount(accountConfig.name);
+  updateDefaultAccount(accountConfig.name!);
   return accountConfig;
-};
-
-const CONFIG_CREATION_FLOWS = {
-  [PERSONAL_ACCESS_KEY_AUTH_METHOD.value]: personalAccessKeyConfigCreationFlow,
-  [OAUTH_AUTH_METHOD.value]: oauthConfigCreationFlow,
-};
+}
 
 const AUTH_TYPE_NAMES = {
   [PERSONAL_ACCESS_KEY_AUTH_METHOD.value]: PERSONAL_ACCESS_KEY_AUTH_METHOD.name,
   [OAUTH_AUTH_METHOD.value]: OAUTH_AUTH_METHOD.name,
 };
 
-exports.command = 'init';
-exports.describe = i18n(`${i18nKey}.describe`, {
+export const command = 'init';
+export const describe = i18n(`${i18nKey}.describe`, {
   configName: DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME,
 });
 
-exports.handler = async options => {
+type InitArgs = CommonArgs &
+  ConfigArgs &
+  TestingArgs & {
+    authType?: string;
+    account?: string;
+    'disable-tracking'?: boolean;
+    'use-hidden-config'?: boolean;
+  };
+
+export async function handler(
+  args: ArgumentsCamelCase<InitArgs>
+): Promise<void> {
   const {
     authType: authTypeFlagValue,
     c: configFlagValue,
     providedAccountId,
     disableTracking,
     useHiddenConfig,
-  } = options;
+  } = args;
   const authType =
     (authTypeFlagValue && authTypeFlagValue.toLowerCase()) ||
     PERSONAL_ACCESS_KEY_AUTH_METHOD.value;
 
   const configPath =
     (configFlagValue && path.join(getCwd(), configFlagValue)) ||
     getConfigPath('', useHiddenConfig);
-  setLogLevel(options);
+  setLogLevel(args);
 
   if (!disableTracking) {
     trackCommandUsage('init', {
       authType,
     });
   }
 
-  const env = options.qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
+  const env = args.qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
 
-  if (fs.existsSync(configPath)) {
+  if (fs.existsSync(configPath!)) {
     logger.error(
       i18n(`${i18nKey}.errors.configFileExists`, {
-        configPath,
+        configPath: configPath!,
       })
     );
     logger.info(i18n(`${i18nKey}.logs.updateConfig`));
     process.exit(EXIT_CODES.ERROR);
   }
 
   if (!disableTracking) {
-    await trackAuthAction('init', authType, TRACKING_STATUS.STARTED);
+    await trackAuthAction(
+      'init',
+      authType,
+      TRACKING_STATUS.STARTED,
+      providedAccountId!
```

Comment:
Same here with using the account you attempted to auth

Reply:
I don't think we can do this because we won't know the account ID until the request for the access token comes back. We'll only know it if the user passes in a `providedAccountId`. So in most cases this will be undefined, but I figured it might be nice to pass it in whenever it does exist. Not sure how much value we'll get out of it though 🤷